src/AlevinUtils.cpp

```diff
@@ -378,21 +378,35 @@ namespace alevin {
         (bc.assign(read, pt.umiLength, pt.barcodeLength), true) : false;
     }
     template <>
-    bool extractBarcode<apt::InDrop>(std::string& read, 
+    bool extractBarcode<apt::InDropV2>(std::string& read, 
                                      std::string& read2, 
-                                     apt::InDrop& pt, std::string& bc){
-      (void)read2;
-      std::string::size_type index = read.find(pt.w1);
-      if (index == std::string::npos){
-        return false;
+                                     apt::InDropV2& pt, std::string& bc){
+      (void)read;
+      if(read2.length() >= (pt.w1Length + pt.barcodeLength + pt.umiLength)) {
+      pt.w1Pos = read2.find(pt.w1);
+      if (pt.w1Pos == std::string::npos){
+        bool found = false;
+        for( int i = 8; i <= 11; i++){
+          if (hammingDistance(pt.w1, read2.substr(i,pt.w1Length)) <= pt.maxHammingDist) {
```

Comment:
@Gaura :  I think this current behavior is OK, but we should be careful to document it.  Specifically, the approach to finding a barcode as implemented here seems to be:

Start at offset 8 and scan through offset 11, accepting the *first* substring whose Hamming distance to the target `pt.w1` is within the required bound.  This means if there are multiple indices that achieve this bound, we return at the first one found.  Likewise, if one has a distance of 2 but a later one has a distance of 1 or 0, we will return the 2.  This seems an unlikely scenario, but is possible and probably should, at least, be documented.

Reply:
Great suggestion. I'll make a separate PR with the updated documentation which will include updates on other protocols as well.